src/folio/service-points-api.ts

```diff
@@ -1,15 +1,48 @@
 import FolioAPI from "./folio-api.js"
-import { CqlParams, ServicePoint } from '../schema.js'
+import { ServicePoint, ServicePointDetails, CqlParams } from '../schema.js'
 
 interface ServicePointsResponse {
   servicepoints: ServicePoint[]
 }
 
 export default class ServicePointsAPI extends FolioAPI {
+  // Get an individual service point using the id
+  async getServicePoint(id: string): Promise<ServicePoint> {
+    const servicePoint = await this.get<ServicePoint>(`/service-points/${encodeURIComponent(id)}`)
+    return this.addDetails(servicePoint)
+  }
+
+  // Parse the description string and add the resulting JSON object in the details filed
+  addDetails(servicePoint: ServicePoint) {
+    return {
+      details: parseServicePointDetails(servicePoint),
+      ...servicePoint
+    }
+  }
+
+  // Get a list of all service points along with the details information required for paging
   async getServicePoints(params: Partial<{ params: CqlParams, [key: string]: object | object[] | undefined }>): Promise<ServicePoint[]> {
     const urlParams = this.buildCqlQuery(params)
-
     const response = await this.get<ServicePointsResponse>(`/service-points`, { params: urlParams })
-    return response.servicepoints
+    return  response.servicepoints.map(sp => {
+      return this.addDetails(sp)
+    })
+  }
+}
+
+// Parse JSON in the description field into a typed ServicePointDetails object.
+// FOLIO stores this field as a string; we use it to store extra metadata about
+// the service point for requests configuration.
+function parseServicePointDetails(servicePoint: ServicePoint): ServicePointDetails {
+  return parseDescriptionJSON(servicePoint.description)
+}
+
+// If the description field is not empty and can be parsed as JSON, we wish to return the JSON object
+// In case a regular string is added to the description field, we also wish to return it in the "notes" key
+function parseDescriptionJSON(description: string) {
+  try {
+    return description ? JSON.parse(description) as ServicePointDetails : null
+  } catch(e) {
+    return description ? {"notes": description} as ServicePointDetails : null
   }
 }
```

Comment:
is there a scenario where we call either of these separately? imo they can be combined; I'd just add all of `parseDescriptionJSON` to `parseServicePointDetails`

Reply:
Updated this